// @anandmohankp/shared/bundles/anandmohankp-shared.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/router')) :
    typeof define === 'function' && define.amd ? define('@anandmohankp/shared', ['exports', '@angular/core', '@angular/router'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.anandmohankp = global.anandmohankp || {}, global.anandmohankp.shared = {}), global.ng.core, global.ng.router));
})(this, (function (exports, i0, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    var SharedService = /** @class */ (function () {
        function SharedService() {
        }
        SharedService.prototype.getSharedServiceContent = function () {
            console.log("****========--");
            return 'version two';
        };
        return SharedService;
    }());
    SharedService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0__namespace, type: SharedService, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    SharedService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0__namespace, type: SharedService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0__namespace, type: SharedService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root'
                    }]
            }], ctorParameters: function () { return []; } });

    var SharedComponent = /** @class */ (function () {
        function SharedComponent() {
            console.log("---");
        }
        SharedComponent.prototype.ngOnInit = function () {
        };
        return SharedComponent;
    }());
    SharedComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0__namespace, type: SharedComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    SharedComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.1.5", type: SharedComponent, selector: "lib-shared", ngImport: i0__namespace, template: "\n    <p>\n      shared works!\n    </p>\n  ", isInline: true });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0__namespace, type: SharedComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'lib-shared',
                        template: "\n    <p>\n      shared works!\n    </p>\n  ",
                        styles: []
                    }]
            }], ctorParameters: function () { return []; } });

    var routes = [{ path: '', component: SharedComponent }];
    var routerModule = i1.RouterModule.forChild(routes);
    var SharedModule = /** @class */ (function () {
        function SharedModule() {
        }
        return SharedModule;
    }());
    SharedModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0__namespace, type: SharedModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    SharedModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0__namespace, type: SharedModule, declarations: [SharedComponent], imports: [i1__namespace.RouterModule], exports: [SharedComponent] });
    SharedModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0__namespace, type: SharedModule, imports: [[
                routerModule
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0__namespace, type: SharedModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [
                            SharedComponent
                        ],
                        imports: [
                            routerModule
                        ],
                        exports: [
                            SharedComponent
                        ]
                    }]
            }] });

    /*
     * Public API Surface of shared
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.SharedComponent = SharedComponent;
    exports.SharedModule = SharedModule;
    exports.SharedService = SharedService;
    exports.routerModule = routerModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
