// @anandmohankp/shared/esm2015/lib/shared.module.js
import { NgModule } from '@angular/core';
import { SharedComponent } from './shared.component';
import { RouterModule } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
const routes = [{ path: '', component: SharedComponent }];
export const routerModule = RouterModule.forChild(routes);
export class SharedModule {
}
SharedModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0, type: SharedModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SharedModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0, type: SharedModule, declarations: [SharedComponent], imports: [i1.RouterModule], exports: [SharedComponent] });
SharedModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0, type: SharedModule, imports: [[
            routerModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.1.5", ngImport: i0, type: SharedModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        SharedComponent
                    ],
                    imports: [
                        routerModule
                    ],
                    exports: [
                        SharedComponent
                    ]
                }]
        }] });
